models/repo/repo.go

```diff
@@ -672,6 +672,44 @@ func GetRepositoryByName(ownerID int64, name string) (*Repository, error) {
 	return repo, err
 }
 
+func GetRepositoryByURL(repoURL string) (*Repository, error) {
+	// possible urls for git:
+	//  https://my.domain/sub-path/<owner>/<repo>.git
+	//  git+ssh://user@my.domain/<owner>/<repo>.git
+	//  user@my.domain:<owner>/<repo>.git
+
+	retrievePathSegments := func(repoURL string) []string {
```

Comment:
Would move this to a standalone function instead of an immediately called anonymous one.